### PR TITLE
Test project template against Django's branches

### PR DESCRIPTION
**Pull request by laymonage**
This ensures our project template does not raise any warnings when used with the latest version of Django and Python.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -153,6 +153,24 @@ jobs:
...
+          wagtail start testproject
+          cd testproject
+          # Remove when all experimental jobs use Django >= 6.0
+          tee -a testproject/settings/local.py << EOF
+          from warnings import filterwarnings
+          filterwarnings(
+              "ignore", "The FORMS_URLFIELD_ASSUME_HTTPS transitional setting is deprecated."
+          )
+          FORMS_URLFIELD_ASSUME_HTTPS = True
+          EOF
```
Comment (laymonage) [inline .github/workflows/test.yml:170]: This is awkward, but necessary for now. Adam Johnson wrote an explanation in his blog post: https://adamj.eu/tech/2023/12/07/django-fix-urlfield-assume-scheme-warnings

File: wagtail/migrations/0087_alter_grouppagepermission_unique_together_and_more.py [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+)
+
+check_constraint_kwargs = {"name": "permission_or_permission_type_not_null"}
+
+if DJANGO_VERSION >= (5, 1):
+    check_constraint_kwargs["condition"] = check_constraint_condition
+else:
+    check_constraint_kwargs["check"] = check_constraint_condition
+
+
```
Comment (Stormheg) [inline wagtail/migrations/0087_alter_grouppagepermission_unique_together_and_more.py:20]: ❤️ 

One less is warning polluting my test logs 😄

**Review by gasman — APPROVED** [review]
LGTM!

**Comment by gasman** [comment]
Merged in 252ca973d3e69ed4e950715c225c016c21bf1561

**Pull request closed without merging**